src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -222,6 +262,9 @@ class RequiredMandatoryAttribute:
                                               problem=f"Failed validation of value on {location.as_string(self.cluster_mapper)}: {str(e)}", spec_location="Global Elements")
                             success = False
                             continue
+                        except KeyError:
+                            # This would have been caught already in the previous step
```

Comment:
Not sure I follow this comment, if it would have been caught already, is code ever expected to end up here? If not why include this?

Reply:
It doesn't assert - it just logs a problem and moves on. We don't need to re-log the problem. This happens if the global attribute is missing. If it's missing, we don't need to check if it's in range. The problem of the missing global attribute happens in the section above.

Reply:
made the comment more verbose.